### PR TITLE
[web] run all text layout benchmarks in CanvasKit mode

### PR DESCRIPTION
**Pull request by yjbanov**
Add CanvasKit mode to `BenchTextLayout` and `BenchTextCachedLayout` benchmarks.

### CODE REVIEW

File: dev/benchmarks/macrobenchmarks/lib/src/web/bench_text_layout.dart [hunk 1]
```diff
@@ -47,30 +47,60 @@ class ParagraphGenerator {
...
+
 /// Sends a platform message to the web engine to enable/disable the usage of
-/// the new canvas-based text measurement implementation.
-void _useCanvasText(bool useCanvasText) {
+/// the canvas-based text measurement implementation.
+void _setTestMode(_TestMode mode) {
+  bool useCanvasText; // null means do not force DOM or canvas, works for CanvasKit
+  switch (mode) {
+    case _TestMode.useDomTextLayout:
+      useCanvasText = true;
```
Comment (mdebbar) [inline dev/benchmarks/macrobenchmarks/lib/src/web/bench_text_layout.dart:68]: This should be false, no?

File: dev/benchmarks/macrobenchmarks/lib/src/web/bench_text_layout.dart [hunk 2]
```diff
@@ -47,30 +47,60 @@ class ParagraphGenerator {
...
+      break;
+    case _TestMode.useCanvasTextLayout:
+      useCanvasText = false;
```
Comment (mdebbar) [inline dev/benchmarks/macrobenchmarks/lib/src/web/bench_text_layout.dart:71]: This is also reversed.

**Review by mdebbar — APPROVED** [review]

**Review by harryterkelsen — APPROVED** [review]
LGTM if `_setTestMode` is fixed (see Mouad's comments)

File: dev/benchmarks/macrobenchmarks/lib/src/web/bench_text_layout.dart (hunk 1 above)
Comment (yjbanov) [reply dev/benchmarks/macrobenchmarks/lib/src/web/bench_text_layout.dart]: Oops! Fixed. Nice catch.

File: dev/benchmarks/macrobenchmarks/lib/src/web/bench_text_layout.dart (hunk 2 above)
Comment (yjbanov) [reply dev/benchmarks/macrobenchmarks/lib/src/web/bench_text_layout.dart]: Done.

**Pull request merged**